pkg/deploy-manager/common.go

```diff
@@ -105,32 +160,79 @@ func (t *DeployManager) GetDeploymentImage(name string) (string, error) {
 }
 
 // WaitForPVCBound waits for a pvc with a given name and namespace to reach BOUND phase
-func (t *DeployManager) WaitForPVCBound(pvcName string, pvcNamespace string) {
-	gomega.Eventually(func() error {
-		pvc, err := t.k8sClient.CoreV1().PersistentVolumeClaims(pvcNamespace).Get(pvcName, metav1.GetOptions{})
-		if err != nil {
-			return err
+func (t *DeployManager) WaitForPVCBound(pvc *k8sv1.PersistentVolumeClaim, namespace string) error {
+	pvc, err := t.k8sClient.CoreV1().PersistentVolumeClaims(namespace).Create(pvc)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return err
+	}
+
+	lastReason := ""
+	timeout := 100 * time.Second
+	interval := 1 * time.Second
+
+	// Wait for namespace to terminate
+	err = utilwait.PollImmediate(interval, timeout, func() (done bool, err error) {
+		pvc, err := t.k8sClient.CoreV1().PersistentVolumeClaims(pvc.Namespace).Get(pvc.Name, metav1.GetOptions{})
+		if err != nil && !errors.IsNotFound(err) {
+			lastReason = fmt.Sprintf("error talking to k8s apiserver: %v", err)
+			return false, nil
 		}
 
-		if pvc.Status.Phase == k8sv1.ClaimBound {
-			return nil
+		if pvc.Status.Phase != k8sv1.ClaimBound {
+			lastReason = fmt.Sprintf("waiting on pvc %s/%s to reach bound state, currently %s", pvc.Namespace, pvc.Name, pvc.Status.Phase)
+			return false, nil
 		}
-		return fmt.Errorf("Waiting on pvc %s/%s to reach bound state when it is currently %s", pvcNamespace, pvcName, pvc.Status.Phase)
-	}, 200*time.Second, 1*time.Second).ShouldNot(gomega.HaveOccurred())
+
+		return true, nil
+	})
+
+	if err != nil {
+		return fmt.Errorf("%v: %s", err, lastReason)
+	}
+
+	return nil
 }
 
 // WaitForJobSucceeded waits for a Job with a given name and namespace to succeed until 200 seconds
-func (t *DeployManager) WaitForJobSucceeded(jobName string, jobNamespace string) {
-	gomega.Eventually(func() error {
-		job, err := t.k8sClient.BatchV1().Jobs(jobNamespace).Get(jobName, metav1.GetOptions{})
-		if err != nil {
-			return err
+func (t *DeployManager) WaitForJobSucceeded(job *batchv1.Job, namespace string) error {
+	job, err := t.k8sClient.BatchV1().Jobs(namespace).Create(job)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return err
+	}
+
+	lastReason := ""
+	timeout := 200 * time.Second
+	interval := 1 * time.Second
```

Comment:
Do we need to retry every second? Can't we get away with every minute?
I don't suppose this job will complete within 1 or 2 or 5 seconds.

And, having a hard-coded 200 seconds timeout doesn't look right. We can't be waiting for 200s for each job.
Can we make it a parameter and pass some realistic values? it'll drastically reduce our test run time.